src/shared/type.ts

```diff
@@ -0,0 +1,60 @@
+interface Reaction {
+  likes: number
+  dislikes: number
+}
+
+export interface Post {
+  author: User
+  id: number
+  title: string
+  body: string
+  tags: string[]
+  reactions: Reaction
+  views: number
+  userId: number
+}
+
+export interface PostsRes {
+  posts: Post[]
+  total: number
+  skip: number
+  limit: number
+}
+
+export interface Comment {
+  id: number
+  body: string
+  postId: number
+  likes: number
+  user: {
+    id: number
+    username: string
+    fullName: string
+  }
+}
+
+export interface CommentsRes {
+  comments: Comment[]
+  total: number
+  skip: number
+  limit: number
+}
+
+export interface User {
+  id: number
+  username: string
+  image: string
+}
```

Comment:
유저(User)의 정보를 받아올 때 쓰는 값과 게시물(Post)의 글쓴이(Author)에서 쓰는 값의 프로퍼티가 서로 다르더라구요ㅎㅎ

그래서 Author와 User를 분리해서 관리하는 것도 방법일 것 같습니당!

```suggestion
export interface Author {
  id: number
  username: string
  image: string
}

export interface User {
// … 생략
}
```

Reply:
확인 감사합니다:) !!!